cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJob.java

```diff
@@ -267,6 +268,11 @@ public void run(RuntimeJobEnvironment runtimeJobEnv) throws Exception {
       try (Program program = createProgram(cConf, programRunner, programDescriptor, programOpts)) {
         ProgramController controller = programRunner.run(program, programOpts);
         controllerFuture.complete(controller);
+        runtimeClientService.acceptStopProgramCommandAsync(isGracefulStopRequested -> {
+          if (isGracefulStopRequested) {
+            controller.stop();
```

Comment:
so if it is not a graceful stop, then what is it? And why don't stop if it is not graceful??